interface/frontend/src/components/open-image/OpenDICOM.vue

```diff
@@ -0,0 +1,122 @@
+<template>
+  <div class="DICOM-container">
+    <div class="DICOM-description">{{ display }}</div>
+    <div class="DICOM" ref="DICOM"></div>
+  </div>
+</template>
+
+<script>
+  const cornerstone = require('cornerstone-core')
+
+  export default {
+    name: 'open-dicom',
+    props: {
+      view: {
+        type: Object,
+        default: {
+          type: 'DICOM',
+          prefixCS: ':/',
+          prefixUrl: '',
+          path: ''
+        }
+      }
+    },
+    data () {
+      return {
+        base64data: null
+      }
+    },
+    computed: {
+      info () {
+        return this.$axios.get(this.view.prefixUrl + this.view.path)
+          .then((response) => {
+            return response.data
+          })
+          .catch(() => {
+            // TODO: handle error
+          })
+      },
+      dicom () {
+        return this.info.then((info) => {
+          this.base64data = info.image
+          return {
+            imageId: this.view.type + this.view.prefixCS + this.view.path,
+            slope: info.metadata['Rescale Slope'],
+            rows: info.metadata['Rows'],
+            columns: info.metadata['Columns'],
+            height: info.metadata['Rows'],
+            width: info.metadata['Columns'],
+            columnPixelSpacing: info.metadata['Pixel Spacing']['0'],
+            rowPixelSpacing: info.metadata['Pixel Spacing']['1'],
+            sizeInBytes: info.metadata['Rows'] * info.metadata['Columns'] * 2,
+            minPixelValue: 0,
+            maxPixelValue: 255,
+            intercept: 0,
+            windowCenter: 110,
+            windowWidth: 100,
+            render: cornerstone.renderGrayscaleImage,
+            getPixelData: this.getPixelData,
+            color: false
+          }
+        })
+      },
+      display () {
+        return this.dicom.then((dicom) => {
```

Comment:
For fancy async/await (it's shipped on latest chromium), we can also do something like this:

```
async display () {
   const element = this.$refs.DICOM
   this.initCS(element)
          
   cornerstone.registerImageLoader(this.view.type, this.dicom)
         
    const image = await cornerstone.loadImage(dicom.imageId)

    cornerstone.displayImage(element, image) 
}
```

Probably need to experiment with it a lil bit more. But overall, LGTM!

Reply:
Thanks, I've read about it, and implement by [this](https://github.com/concept-to-clinic/concept-to-clinic/pull/188/commits/64484a2ed46aedb1d3fdc5ff58c0cdd344289c6a).

Reply:
Sweet :ok_hand: